File: src/_eu/ml/Audio/AudioMenu.js
/******************************************************************************
 * Copyright 2014-2016 Vladimir Tolmachev
 * Copyright 2016 Visionarity AG
 * Vladimir Tolmachev and Visionarity AG have unlimited commercial
 * licenses for commercial use and customization
 *
 * Author: Vladimir Tolmachev (tolm_vl@hotmail.com)
 * Ported C++ to Javascript: Visionarity AG / Vladimir Tolmachev
 * Project: Island Defense (JS)
 * If you received the code not from the author, please contact us
 ******************************************************************************/

//Define namespace
var EU = EU || {};

EU.AudioMenu = cc.Menu.extend(
{
    m_soundOn : null,
    m_soundOff : null,
    m_musicOn : null,
    m_musicOff : null,
    //CC_SYNTHESIZE(MenuItemSprite *, m_soundOn, SoundOn);
    //CC_SYNTHESIZE(MenuItemSprite *, m_soundOff, SoundOff);
    //CC_SYNTHESIZE(MenuItemSprite *, m_musicOn, MusicOn);
    //CC_SYNTHESIZE(MenuItemSprite *, m_musicOff, MusicOff);

    ctor: function()
    {
        this._super();
        var desSize = cc.view.getDesignResolutionSize();

        this.setPosition( desSize /2, desSize/2 );

        var cb0 = EU.AudioEngine.soundEnabled.bind(this, false);
        var cb1 = function() { EU.AudioEngine.soundEnabled(true);  EU.AudioEngine.playEffect2( "" ); } ;
        var cb2 = function() { EU.AudioEngine.musicEnabled(false); EU.AudioEngine.playEffect2( "" ); } ;
        var cb3 = function() { EU.AudioEngine.musicEnabled(true);  EU.AudioEngine.playEffect2( "" ); } ;

        this.m_soundOn = new cc.MenuItemImage( "",  "", cb0 );
        this.m_soundOff = new cc.MenuItemImage( "", "", cb1 );
        this.m_musicOn = new cc.MenuItemImage( "", "", cb2 );
        this.m_musicOff = new cc.MenuItemImage( "", "", cb3 );

        this.addChild( this.m_soundOn );
        this.addChild( this.m_soundOff );
        this.addChild( this.m_musicOn );
        this.addChild( this.m_musicOff );

        var K = 75;
        this.m_soundOn.setPosition( Point(desSize.width - K*2, desSize.height - K) );
        this.m_soundOff.setPosition( Point(desSize.width - K*2, desSize.height - K) );
        this.m_musicOn.setPosition( Point(desSize.width - K, desSize.height - K) );
        this.m_musicOff.setPosition( Point(desSize.width - K, desSize.height - K) );

    },

    check: function()
    {
        //var s = UserDefault.getInstance().getBoolForKey( "", true );
        //var m = UserDefault.getInstance().getBoolForKey( "", true );

        //this.m_soundOn.setVisible ( s );
        //this.m_soundOff.setVisible( !s );
        //this.m_musicOn.setVisible ( m );
        //this.m_musicOff.setVisible( !m );

    }
});

File: src/_cpp/ml/Audio/AudioMenu.cpp
//
//  AudioMenu.cpp
//  TowerDefence
//
//  Created by Vladimir Tolmachev on 04.02.14.
//
//

#include "SimpleAudioEngine.h"
#include "ml/Audio/AudioMenu.h"
#include "ml/Audio/AudioEngine.h"
NS_CC_BEGIN;
using namespace CocosDenshion;

AudioMenu::AudioMenu()
{
	auto desSize = Director::getInstance()->getOpenGLView()->getDesignResolutionSize();
	
	Menu::init();
	setPosition( Point::ZERO );

	auto cb0 = std::bind( [](){ AudioEngine::shared().soundEnabled(false); } );
	auto cb1 = std::bind( [](){ AudioEngine::shared().soundEnabled(true);  AudioEngine::shared().playEffect2( "" ); } );
	auto cb2 = std::bind( [](){ AudioEngine::shared().musicEnabled(false); AudioEngine::shared().playEffect2( "" ); } );
	auto cb3 = std::bind( [](){ AudioEngine::shared().musicEnabled(true);  AudioEngine::shared().playEffect2( "" ); } );

	m_soundOn = MenuItemImage::create ( "",  "", cb0 );
	m_soundOff = MenuItemImage::create( "", "", cb1 );
	m_musicOn = MenuItemImage::create( "", "", cb2 );
	m_musicOff = MenuItemImage::create( "", "", cb3 );

	addChild( m_soundOn );
	addChild( m_soundOff );
	addChild( m_musicOn );
	addChild( m_musicOff );

	float K(75);
	m_soundOn->setPosition( Point(desSize.width - K*2, desSize.height - K) );
	m_soundOff->setPosition( Point(desSize.width - K*2, desSize.height - K) );
	m_musicOn->setPosition( Point(desSize.width - K, desSize.height - K) );
	m_musicOff->setPosition( Point(desSize.width - K, desSize.height - K) );

}

AudioMenu::~AudioMenu()
{
}

void AudioMenu::check()
{
	//bool s = UserDefault::getInstance()->getBoolForKey( "", true );
	//bool m = UserDefault::getInstance()->getBoolForKey( "", true );

	//m_soundOn->setVisible ( s );
	//m_soundOff->setVisible( !s );
	//m_musicOn->setVisible ( m );
	//m_musicOff->setVisible( !m );
	
}


NS_CC_END;